packages/tlinks/src/ui/FarcasterFrame.tsx

```diff
@@ -0,0 +1,202 @@
+"use client";
+import {
+  useState,
+  useMemo,
+  useCallback,
+} from "react";
+import { useFarcasterIdentity } from "@frames.js/render/identity/farcaster";
+import { useFrame } from "@frames.js/render/use-frame";
+import {
+  fallbackFrameContext,
+  type OnTransactionArgs,
+  signFrameAction,
+} from "@frames.js/render";
+import {
+  FrameUI,
+  type FrameUIComponents,
+} from "@frames.js/render/ui";
+import { WebStorage } from "@frames.js/render/identity/storage";
+import { useNeynarContext } from "@neynar/react";
+import { getBlockExplorerUrl, theme, type FarcasterUser, type StylingProps } from "../utils/constants.ts";
+import type { ActionAdapter } from "../api/ActionConfig.ts";
+import { fetchScriptURI } from "../utils/fetch-ts-data.ts";
+
+interface FarcasterFrameProps {
+  chainId: number;
+  scriptURI?: string;
+  cssClass?: string;
+  farcaster: FarcasterUser
+  adapter: ActionAdapter
+}
+
+const VIEWER_PATTERNS = [
+  "viewer",
+  "viewer-staging",
+  "https://ipfs.io/ipfs",
+  "ipfs://ipfs",
+  "ipfs://",
+];
+
+export async function isFarcasterFrame(
```

Comment:
the actionUrl can be `https://buy-me-a-beer-sigma.vercel.app/TantanFu` for these urls which are not TS viewer link, we can do a early return instead of throw error, that's the expected cases.
Apart from TS viewer links, we also support tlink.store links, please reuse `isTokenScriptViewerUrl`